aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -71,9 +71,12 @@ void replication_pad1d_out_cpu_template(
   int pad_l = paddingSize[0];
   int pad_r = paddingSize[1];
 
-  TORCH_CHECK(input_.numel() > 0
-      && (input_.ndimension() == 2 || input_.ndimension() == 3),
-      "non-empty 2D or 3D (batch mode) tensor expected for input");
+  
+  // allow empty batch size but not other dimensions.
+  TORCH_CHECK(
```

Comment:
If we want to allow only empty batch but not other dimensions, the check here is wrong, should be 
```
auto size_i = input_.size();
TORCH_CHECK((size_i.size()==2 || size_i.size()==3) &&  
std::all_of(size_i.rbegin(), size_i.rbegin()+2, [](auto i){return i!=0;}));
```
This covers both batch mode and non-batch mode. In non-batch mode (2D) you want all values to be non-zero, in the batch mode (3D) you allow only the 0-th element to potentially be 0. The error message also should be clearer - you are erroring out if some of the dimensions are 0, but the error message does not reflect that. 
Same goes for other checks - they all allow more zeros than is right, and all don't provide clear error message.

Reply:
OK this one is wrong and I have corrected it . However, the rest of the checks seem to make sense to me.

* https://github.com/pytorch/pytorch/pull/39137/files#diff-3e39c997ceefe5db9d4b51504dc8c91cR574 checks for dims 1,2,3 for 4D tensor and 1,2,3,4 for 5D tensor.
* This one too: https://github.com/pytorch/pytorch/pull/39137/files#diff-01f6085523cbc69ca71060d98e22f257R364